pennylane/devices/default_mixed.py

```diff
@@ -905,3 +984,77 @@ def execute(
         execution_config: Optional[ExecutionConfig] = None,
     ) -> None:
         raise NotImplementedError
+
+    def _setup_execution_config(self, execution_config: ExecutionConfig) -> ExecutionConfig:
+        """This is a private helper for ``preprocess`` that sets up the execution config.
+
+        Args:
+            execution_config (ExecutionConfig): an unprocessed execution config.
+
+        Returns:
+            ExecutionConfig: a preprocessed execution config.
+        """
+        updated_values = {}
+        for option in execution_config.device_options:
+            if option not in self._device_options:
+                raise qml.DeviceError(f"device option {option} not present on {self}")
+
+        if execution_config.gradient_method == "best":
+            updated_values["gradient_method"] = "backprop"
+        updated_values["use_device_gradient"] = False
```

Comment:
```suggestion
        updated_values["use_device_gradient"] = execution_config.gradient_method in {"backprop", "best"}
```
We want to use the device gradient in these situations.

Reply:
backprop kinda is a special case, but we are still denoting it as a device gradient.